src/tracing/service/tracing_service_impl.cc

```diff
@@ -4461,6 +4531,17 @@ base::Status TracingServiceImpl::FinishCloneSession(
   cloned_session->final_flush_outcome = final_flush_outcome
                                             ? TraceStats::FINAL_FLUSH_SUCCEEDED
                                             : TraceStats::FINAL_FLUSH_FAILED;
+
+  if (output_file_fd) {
+    uint64_t bytes_written_into_file = 0;
+    DoReadBuffersIntoFile(cloned_session, output_file_fd,
```

Comment:
it's not clear to me why here you don't just

1. set the session write_period_ms = 0, which will trigger a stop/flush
2. just call ReadBuffersIntoFile(tsid /*the cloned id*) here